tests/expose_test.go

```diff
@@ -184,13 +293,28 @@ var _ = Describe("[rfe_id:253][crit:medium][vendor:cnv-qe@redhat.com][level:comp
 					Expect(node.Status.Addresses).ToNot(BeEmpty())
 					nodeIP := node.Status.Addresses[0].Address
 
+					if ipFamily == k8sv1.IPv6Protocol {
+						ipv6NodeIP, err := resolveNodeIPAddrByFamily(
```

Comment:
According to the [documentation](https://kubernetes.io/docs/tasks/network/validate-dual-stack/#validate-node-addressing) the `node.Status.Addresses` should contain both `ipv4` and `ipv6` addresses. 

IIUC, [this](https://github.com/kubernetes/kubernetes/pull/95239/files#diff-15a7ea42c856468667780447090b682f36ea0385dcdbe9105ff49900564a190bR101) merged PR is solving the issue.

@oshoval can we please update the cluster?

Reply:
@AlonaKaplan , I do not know if that PR is relevant to this issue, they did spoke about pod HostIP while the function handled the node.
On k8s, there was a long standing test that covered this, see [here](https://github.com/kubernetes/kubernetes/commit/789920955a9cda382d1d4d5557469c881d8c93f7#diff-20f023ec27befdd70ec0f9283ebabd364679d9ab8b7f95ae87449159d58396a5R59).
So I am not clear why the two addresses are not seen today or how that test is passing on k8s.

Perhaps we are not setting or defining the nodes when registering them to the cluster in a way that the dual-stack is detected right?

Reply:
I think that if the [cloud-controller-manager](https://kubernetes.io/docs/concepts/architecture/cloud-controller/) is in place, the two addresses are seen. For bare metal, the ref PR may be a solution, I do not really understand (as it mentions passing the IP addresses manually in the config, which is weird) if this is what is really missing. Seems to me that calico also needs to be in the picture here.

Reply:
There it is: https://github.com/kubevirt/kubevirtci/issues/462